### ISSUE TITLE
Obsolete AutomationProperties properties cause XamlC exception on NET 8

### ISSUE THREAD

**Issue by GalaxiaGuy** [dotnet/maui#17484]
### Description

Using `AutomationProperties.Name` or `AutomationProperties.Hint` (and I would _guess_ the other deprecated `AutomationProperties` attached properties) inside a `DataTemplate` in a control's `Resources` generates the error `XamlC error : Object reference not set to an instance of an object.` at compile time.

I'm in the process of porting a large app. It previously worked against .NET 7 and did not see this issue.

### Steps to Reproduce

Add something like the following to a XAML file:
```xml
    <ContentPage.Resources>
        <DataTemplate x:Key="Template">
            <Image
                Source="dotnet_bot.png"
                AutomationProperties.Name="Cute dot net bot waving hi to you!"
                HeightRequest="200"
                HorizontalOptions="Center" />        
        </DataTemplate>
    </ContentPage.Resources>
```
### Link to public reproduction project repository

https://github.com/GalaxiaGuy/MauiBugs/tree/main

### Version with bug

[8.0.0-preview.1.7762](https://github.com/dotnet/maui/releases/tag/8.0.0-preview.1.7762)

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI, Yes, this used to work in Xamarin.Forms

### Last version that worked well

7.0.92

### Affected platforms

iOS, Android

### Did you find any workaround?

Updating the properties to the correct `SemanticProperties` equivalents allows the project to compile. It took some trial and error to work out what was causing the problem though.

**Comment by samhouts** [comment]
The AutomationProperties were obsoleted in https://github.com/dotnet/maui/pull/13104, which released in [8.0.0-preview.1.7762](https://github.com/dotnet/maui/releases/tag/8.0.0-preview.1.7762). I think it's technically not a regression in XamlC (it probably always had the behavior of blowing up on now non-existent attached properties), but it does need to have a better error message.

**Comment by StephaneDelcroix** [comment]
there is a LogWarning, but it looks like it's throwing anyway
